### PR TITLE
Add workflow to prepare release branch

### PR DESCRIPTION
**Pull request by trask**
Automate a couple of the manual release steps

I tested this on my fork, you can look there to see example of the two PRs that this workflow creates: https://github.com/trask/opentelemetry-java-instrumentation/pulls

### CODE REVIEW

**Review by laurit — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]

File: .github/workflows/prepare-release-branch.yml
Comment (anuraaga) [inline .github/workflows/prepare-release-branch.yml:71]: If wanting to avoid a third party action, might be easy enough to use gh CLI

https://cli.github.com/manual/gh_pr_create

File: .github/workflows/prepare-release-branch.yml
Comment (anuraaga) [inline .github/workflows/prepare-release-branch.yml:75]: FWIW, I was trying to see how delete-branch could possibly work when creating a pull request, and indeed it doesn't

https://github.com/peter-evans/create-pull-request#action-behaviour

It would delete the PR if it had already been merged when running this step. So to prevent confusion, we should delete this field if sticking with the action. But I'd say the CLI command is simple and all we need, this action seems to support some intense workflows :O

https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#how-the-action-works

File: RELEASING.md
Comment (anuraaga) [inline RELEASING.md:18]: Make sure to delete the branch after merging

File: .github/workflows/prepare-release-branch.yml
Comment (anuraaga) [inline .github/workflows/prepare-release-branch.yml:92]: I just noticed the `**/**` CNCF branch protection rule seems to be gone, did we accidentally delete it?

Though either way, even if we can make it work for release branch, this PR creation to `main` looks like it'll be problematic due to CLA check :(

Comment (anuraaga) [reply .github/workflows/prepare-release-branch.yml:92]: Though we do explicitly have the checkbox to let us merge even if status checks aren't all passing. So that could be our workaround

Comment (trask) [reply .github/workflows/prepare-release-branch.yml:92]: we have branch protection on both `main` and `v**`, so CLA check will be a problem for both PRs 😞

Comment (trask) [reply .github/workflows/prepare-release-branch.yml:92]: it looks there are [complexities to approving github-actions bot account for EasyCLA](https://github.com/open-telemetry/community/issues/809#issuecomment-905869941) and so that path could take some time

it seems that the current recommendation is to use a Personal Access Token generated by one of the maintainers: https://github.com/open-telemetry/community/blob/main/docs/using-github-extensions.md#writing-your-github-actions-pipelines

Comment (trask) [reply .github/workflows/prepare-release-branch.yml:92]: reverted back to using github-actions bot for now at least

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**
